spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/support/SimpleTaskRepository.java

```diff
@@ -16,28 +16,44 @@
 
 package org.springframework.cloud.task.repository.support;
 
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
 import org.springframework.cloud.task.repository.TaskExecution;
 import org.springframework.cloud.task.repository.TaskRepository;
+import org.springframework.cloud.task.repository.dao.TaskExecutionDao;
 
 /**
- * {@link TaskRepository} implementation that will log the task execution information.
- *
+ * Records the task execution information to the log and to TaskExecutionDao provided.
  * @author Glenn Renfro
  */
-public class LoggerTaskRepository implements TaskRepository {
-	private final static Logger logger = LoggerFactory.getLogger(LoggerTaskRepository.class);
+public class SimpleTaskRepository implements TaskRepository {
+
+	private final static Logger logger = LoggerFactory.getLogger(SimpleTaskRepository.class);
+
+	private TaskExecutionDao taskExecutionDao;
+
+	public SimpleTaskRepository(TaskExecutionDao taskExecutionDao){
+		this.taskExecutionDao = taskExecutionDao;
+	}
 
 	@Override
 	public void update(TaskExecution taskExecution) {
+		taskExecutionDao.updateTaskExecution(taskExecution);
 		logger.info("Updating: " + taskExecution.toString());
 	}
 
 	@Override
 	public void createTaskExecution(TaskExecution taskExecution) {
+		taskExecutionDao.saveTaskExecution(taskExecution);
 		logger.info("Creating: " + taskExecution.toString());
 	}
+
+
+	/**
+	 * Retrieves the taskExecutionDao associated with this repository.
+	 * @return the taskExecutionDao
+	 */
+	public TaskExecutionDao getTaskExecutionDao() {
```

Comment:
In this case so that the Tests can retrieve the Map repository.  Also will be used for TaskExplorer when using the MapRepository.